From: Quoc Trinh <quoc.trinh.fz@renesas.com>
Subject: [PATCH] rz-cmn: boot V4H FIT with OP-TEE when present

Use the V4H FIT only for the Sparrow Hawk DTB. Fall back to the existing
Image plus external DTB flow when the FIT is not installed.

Upstream-Status: Inappropriate [unified rz-cmn boot flow]
Signed-off-by: Quoc Trinh <quoc.trinh.fz@renesas.com>
---
 include/configs/rz-cmn_env.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/configs/rz-cmn_env.h b/include/configs/rz-cmn_env.h
--- a/include/configs/rz-cmn_env.h
+++ b/include/configs/rz-cmn_env.h
@@ -122,4 +122,9 @@
 	"mmc_do_boot=run mmc_args; run image_select; " \
+		"run fdt_select; " \
+		"if test \"${fdtfile}\" = \"r8a779g3-sparrow-hawk.dtb\" && " \
+			"fatload mmc ${mmcdev}:${mmcpart} ${image_addr} fitImage-sparrow-hawk; then " \
+			"bootm ${image_addr}; " \
+		"fi; " \
 		"fatload mmc ${mmcdev}:${mmcpart} ${image_addr} ${kernel_image}; " \
 		"run fdt_ovrun; " \
 		"booti ${image_addr} - ${dtb_addr}\0"
